src/layman/upgrade/upgrade_v2_0.py

```diff
@@ -267,18 +371,84 @@ def migrate_layers():
-            keys_to_check += ['db', 'wfs']
-        if publ_info['original_data_source'] == 'file':
-            keys_to_check += ['file']
-        assert all('status' not in publ_info[key] for key in keys_to_check), json.dumps(publ_info, indent=2)
+        assert publ_info['layman_metadata']['publication_status'] == 'COMPLETE', json.dumps(publ_info, indent=2)
+        os.rmdir(f"{src_main_path}")
 
         logger.info(f'    Migrate layer {workspace}.{layername} DONE')
 
 
+def migrate_maps():
+    logger.info(f'    Migrate maps')
+
+    query = f'''
+    select w.name, p.name, p.uuid::varchar as uuid
+    from {DB_SCHEMA}.publications p inner join
+         {DB_SCHEMA}.workspaces w on w.id = p.id_workspace left join
+         {DB_SCHEMA}.users u on u.id_workspace = w.id
+    where p.type = %s
+    order by w.name, p.name
+    ;'''
+    maps = db_util.run_query(query, (MAP_TYPE,))
+
+    for workspace, mapname, map_uuid, in maps:
+        # check if publication is not yet migrated
+        publ_info = get_complete_map_info(workspace, mapname)
+        publ_status = publ_info['layman_metadata']['publication_status']
+        assert publ_status in ['COMPLETE', 'INCOMPLETE']
+        if publ_status == 'INCOMPLETE':
+            logger.info(f'    Migrate map {workspace}.{mapname} (uuid={map_uuid})')
+        else:
+            logger.warning(f'    Map {workspace}.{mapname} seems already migrated!')
+            continue
+
+        post_task_kwargs = {
+            'uuid': map_uuid,
+        }
+
+        src_main_path = f"{settings.LAYMAN_DATA_DIR}/workspaces/{workspace}/maps/{mapname}"
+        dst_main_path = f"{settings.LAYMAN_DATA_DIR}/maps/{map_uuid}"
+
+        # Move input files
+        logger.info("      moving input files")
+        src_input_file_path = f"{src_main_path}/input_file/{mapname}.json"
+        dst_input_filepath = f"{dst_main_path}/input_file/{map_uuid}.json"
+        os.makedirs(f"{dst_main_path}/input_file/", exist_ok=True)
+        shutil.move(src_input_file_path, dst_input_filepath)
+        os.rmdir(f"{src_main_path}/input_file")
+
+        # Move thumbnail file
+        logger.info("      moving thumbnail file")
+        src_thumbnail_path = f"{src_main_path}/thumbnail/{mapname}.png"
+        dst_thumbnail_path = f"{dst_main_path}/thumbnail/{map_uuid}.png"
+        if os.path.exists(src_thumbnail_path):
+            os.makedirs(f"{dst_main_path}/thumbnail/", exist_ok=True)
+            shutil.move(src_thumbnail_path, dst_thumbnail_path)
+            os.rmdir(f"{src_main_path}/thumbnail")
+        else:
+            util.run_task_sync(map_refresh_thumbnail, [workspace, mapname], post_task_kwargs)
+            shutil.rmtree(f"{src_main_path}/thumbnail")
+
+        # assert that source keys up to geoserver are OK
+        publ_info = get_complete_map_info(workspace, mapname)
+        assert publ_info['layman_metadata']['publication_status'] == 'COMPLETE', json.dumps(publ_info, indent=2)
```

Comment:
I am not sure if there are any maps without thumbnail on 157. If there are any, I think this check will fail for them.